Documentation cleanup for the Hibernate Search extension

yrodiere opened the pull request.

yrodiere committed:
Use @ConfigDocMapKey for index config in Hibernate Search extension

yrodiere committed:
Document default/required values in the Hibernate Search extension

yrodiere committed:
Remove explicit listings of enum values in the Hibernate Search extension

These are already documented automatically by the documentation
generation tool.

yrodiere committed:
Clarify that the "searchable" strategy in the Hibernate Search extension is not a good fit for production

gsmet reviewed (approved):
Looks good, thanks. Let's wait for CI.

yrodiere committed:
Avoid warnings when discovery is disabled in the Hibernate Search extension

Without this change, we get warnings like this:

2019-11-05 10:42:34,385 WARN  [org.hib.sea.eng.cfg.spi.ConfigurationPropertyChecker] (main)
HSEARCH000568: Some properties in the Hibernate Search configuration were not used;
there might be misspelled property keys in your configuration.
Unused properties were:
[hibernate.search.backends._quarkus_.discovery.refresh_interval, hibernate.search.backends._quarkus_.discovery.default_scheme].
To disable this warning, set the property
'hibernate.search.hibernate.search.configuration_property_checking.strategy' to 'ignore'.
